code/modules/art/paintings.dm

```diff
@@ -191,7 +197,39 @@
 	painting_metadata.patron_ckey = user.ckey
 	painting_metadata.patron_name = user.real_name
 	painting_metadata.credit_value = offer_amount
-	to_chat(user, span_notice("Nanotrasen Trust Foundation thanks you for your contribution. You're now an official patron of this painting."))
+	last_patron = WEAKREF(user.mind)
+	to_chat(user, span_notice("Nanotrasen Trust Foundation thanks you for your contribution. You're now offical patron of this painting."))
+	var/list/possible_frames = SSpersistent_paintings.get_available_frames(offer_amount)
+	if(possible_frames.len <= 1) // Not much room for choices here.
+		return
+	if(tgui_alert(user, "Do you want to change the frame appearance now? You can do so later this shift with Alt-Click as long as you're a patron.","Patronage Frame",list("Yes","No")) != "Yes")
```

Comment:
Is this supposed to work only in the round it was picked or anytime later ?

Reply:
It depends on what you're actually asking.
If it's the ability to change the frame, then yes. The reason why it doesn't carry over between rounds is that there's no good way to have it do that without assigning it to the ckey (Urist McPlayer) rather than the character who offered the patronage (Urist McSpessman). This doesn't mean I can't do it, but simply that doing it would expose the feature to the subtle absurdity of someone that isn't Urist McSpessman being able to change the frame because their ckey is that of Urist McPlayer. This could be fixed, albeit only partially, by adding the condition that the patron name matches the mob's real name, though it's counterintuitive since it can also leave Urist McPlayer wondering why they can't access the feature if Urist McSpessman was renamed to something else like Urist McMulligan or Urist McClown.

If instead you're talking about the frame appearance, no. It persists between rounds.

Reply:
Ah i see, that's fair i was thinking of this as more of OOC action so i haven't considered the ckey != mob issue.